Status bars

AdamHostin opened the pull request:
- bar prefab
- usage demonstration on Unit

AdamHostin committed:
ADD: enemy can attack IDamagable objects

AdamHostin committed:
FIX: Enemies check their awareness radius on spawn

AdamHostin committed:
Merge branch 'dev' into feature/status-bars

AdamHostin committed:
ADD: status bars

AdamHostin committed:
ADD: Gradient in bar

AdamHostin committed:
FIX: Enemy colls layer

KubkoFrano committed:
Merge branch 'feature/enemies-attacks-units' into feature/status-bars

AdamHostin committed:
FIX: Enemies are not getting stuck in attack stat

AdamHostin committed:
FIX: minnor fixes (check description)

- enemies don't attack next target from other side of the map
- enemies checks if recently placed unit is in their awareness radius

KubkoFrano commented on Assets/Scripts/Models/Unit.cs:
```diff
@@ -96,22 +102,25 @@ public bool Shoot()
             toxicityResistance -= infection;
             //Debug.Log("Unit infection: " + toxicityResistance + " Unit hp: " + hp);
+            behaviour.hpBar.OnUIUpdate(((float)hp / maxHp),hp,maxHp);
+            behaviour.toxicityBar.OnUIUpdate(((float)(maxToxicityResistance - toxicityResistance + Mathf.Epsilon ) / maxToxicityResistance),  toxicityResistance, maxToxicityResistance);
```
Delete Mathf.Epsilon

KubkoFrano commented on Assets/Scripts/UI/Bar/BarController.cs:
```diff
@@ -13,12 +13,16 @@ public class BarController : MonoBehaviour
     [SerializeField] bool isTextVisible;
     [SerializeField] bool isXpBar;
+    [SerializeField] string postFix;
+    [SerializeField] float Speed;
```
s

AdamHostin committed:
FIX: xp array overflow

AdamHostin committed:
FIX: typo

AdamHostin committed:
FIX: layer bug when multiple enemies are attacking

- speed typo fixed in barcontroller
- spawning 25 enemies
- don't change layer order!!!!!

KubkoFrano reviewed (changes requested):
Pekné riešenie, ešte bude treba pridať do kamery na CameraIgnoreLayers komponent ten enemy layer, ak naňho chceme vedieť klikať

KubkoFrano committed:
ADD: Enemy layer to eventMask
